package/Dockerfile

```diff
@@ -32,7 +15,7 @@ RUN zypper refresh && \
     rm -rf /chroot/var/cache/zypp/* /chroot/var/log/zypp/* /chroot/tmp/* /chroot/var/tmp/* /chroot/usr/share/doc/packages/*
 
 # Main stage using bci-micro as the base image
-FROM final
+FROM scratch AS base
```

Comment:
This was mostly an optimisation. The contents of `bci-micro` is already being copied on `L21`, which means that some files end up duplicated in image final layers:

Using `FROM final as base`:
```
Potential wasted space: 11 MB                          
Image efficiency score: 99 %                           
                                                       
Count   Total Space  Path                              
    2        6.2 MB  /usr/lib/sysimage/rpm/Packages.db 
    2        3.9 MB  /usr/lib/sysimage/rpm/Index.db    
    2        849 kB  /usr/bin/which                    
    2        9.4 kB  /var/cache/ldconfig/aux-cache     
    2        7.5 kB  /etc/ld.so.cache                  
    2        1.4 kB  /var/lib/zypp/AutoInstalled       
    2         271 B  /etc/group                        
    2         108 B  /etc/passwd                       
    2          67 B  /etc/shadow                       
    2          21 B  /etc/subuid                       
    2          21 B  /etc/subgid                       
    2           0 B  /usr/local/bin
```

Using `FROM scratch as base`:
```
Potential wasted space: 849 kB      
Image efficiency score: 99 %        
                                
Count   Total Space  Path           
    2        849 kB  /usr/bin/which 
    2         108 B  /etc/passwd    
    2          67 B  /etc/shadow    
    2          21 B  /etc/subuid    
    2          21 B  /etc/subgid    
    2           0 B  /usr/bin/helm  
    2           0 B  /usr/local/bin
```

@macedogm should I revert this to `bci-micro`?

Reply:
Personally, I would keep bci-micro instead of scratch. That is a personal preference mostly due to:
1. Reduce potential side effects with copy into scratch; and
2. It's unclear, only from the comment, if the files in `/usr/lib/sysimage/rpm/` are removed or kept. If they are kept, then okay, but if they are removed, then that could mean that CVE scanners cannot identify the installed RPM packages inside the image.

Could you confirm if in 2 those files are kept in the final scratch image?

The switch in 1 could also be done on a follow-up PR, as you said, we reduce the amount of big changes from one shoot into multiples.

Reply:
The latest version of such files are kept. The efficiency score tracks files that were modified which results on them being duplicated across 2 or more layers. AFAIK, modification can be at two levels, content or metadata (i.e. file mode).

So when we do `FROM final as base`, we start our final image with a copy of `/usr/lib/sysimage/rpm/Packages.db`, which is latter overwritten by the version updated by the zypper operation. Note that this is only important on the final or in chained `FROM` layers, else such inefficiencies are discarded.

As we are copying the contents of `final` to the root of `scratch`, things that we could lose from `final` in that operation could be environment variables, which in this case are trivial, or Image labels.

Reply:
@macedogm really well spotted, I spent some time on this and I agree we should revert this - which is now done.

As a follow-up, the clean up PR will overwrite the labels that BCI have introduced in the Rancher Manager images which makes it quite confusing for consumers - as it states the current image is BCI/SLE:
```
"org.opencontainers.image.description": "A micro environment for containers based on the SLE Base Container Image.",
"org.opencontainers.image.ref.name": "15.6.24.2",
"org.opencontainers.image.source": "https://sources.suse.com/SUSE:SLE-15-SP6:Update:CR/micro-image/c4136c6519a0f7f58d2c0ca542aee82a/",
"org.opencontainers.image.title": "SLE BCI 15 SP6 Micro",
"org.opencontainers.image.url": "https://www.suse.com/products/base-container-images/",
"org.opencontainers.image.version": "15.6.24.2",
"org.opensuse.reference": "registry.suse.com/bci/bci-micro:15.6.24.2"
```
But let's deal with it separately.